bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/console/HueCommandExtension.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.hue.internal.console;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingRegistry;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.io.console.Console;
+import org.eclipse.smarthome.io.console.extensions.AbstractConsoleCommandExtension;
+import org.eclipse.smarthome.io.console.extensions.ConsoleCommandExtension;
+import org.openhab.binding.hue.internal.handler.HueBridgeHandler;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * The {@link HueCommandExtension} is responsible for handling console commands
+ *
+ * @author Laurent Garnier - Initial contribution
+ */
+
+@NonNullByDefault
+@Component(service = ConsoleCommandExtension.class)
+public class HueCommandExtension extends AbstractConsoleCommandExtension {
+
+    private static final String USER_NAME = "username";
+
+    private final ThingRegistry thingRegistry;
+
+    @Activate
+    public HueCommandExtension(final @Reference ThingRegistry thingRegistry) {
+        super("hue", "Interact with the hue binding.");
+        this.thingRegistry = thingRegistry;
+    }
+
+    @Override
+    public void execute(String[] args, Console console) {
+        if (args.length == 2) {
+            HueBridgeHandler bridgeHandler = null;
+            try {
+                ThingUID thingUID = new ThingUID(args[0]);
+                Thing thing = thingRegistry.get(thingUID);
+                if (thing != null) {
+                    ThingHandler thingHandler = thing.getHandler();
+                    if (thingHandler instanceof HueBridgeHandler) {
+                        bridgeHandler = (HueBridgeHandler) thingHandler;
+                    }
+                }
+            } catch (IllegalArgumentException e) {
+                bridgeHandler = null;
+            }
+            if (bridgeHandler == null) {
+                console.println("Bad bridge id '" + args[0] + "'");
```

Comment:
The lack of a handler doesn't mean that the provided bridge ID was a bad one.

Reply:
You should log something different depending on if the handler was null vs the thing was null.

Reply:
I am a little surprised because this comment comes after few other PRs with the same code you recently reviewed without any such comment.
What would be the case with a null handler ? If the bundle is stopped for example ? What could be the clear message to display to the user ?

Reply:
If the bundle is stopped, the console command will not be accessible. 
The case you ask to handle is probably only if the command is run during the startup of the binding, before the thing handler is initialized. Very low probable to happen in practice.

Reply:
Well you can always disable things without stopping the bundle.
The only reason that I pointed this change out is because I'm a better reviewer than I was yesterday. 🙂

Reply:
I will have to update other bindings in this case,

Reply:
Ok, changed.
In fact, there are 3 different cases, so 3 different messages.